public/scripts/firebase.js

```diff
@@ -378,17 +379,24 @@ friendlyPix.Firebase = class {
       console.error(e);
     }
 
-    const updateData = {
-      profile_picture: imageUrl || null,
-      full_name: displayName,
-      _search_index: {
-        full_name: searchFullName,
-        reversed_full_name: searchReversedFullName
-      }
-    };
-    return this.database.ref(`/people/${this.auth.currentUser.uid}`).update(updateData).then(() => {
-      console.log('Public profile updated.');
-    });
+    friendlyPix.firebase.getPrivacySettings(user.uid).then(snapshot => {
+      const socialEnabled = snapshot.val().social;
```

Comment:
IT seems that for first-sign-in users `snapshot.val()` is `null`. Need a case for this.